### ISSUE TITLE
Provider Plan to adapt to AWS SDK for Go V1 End-of-Support

### ISSUE THREAD

**Issue by breathingdust** [hashicorp/terraform-provider-aws#35642]
### Description

AWS have announced the End-Of-Support plan for the AWS SDK for Go v1:

- https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-aws-sdk-for-go-v1-on-july-31-2025/

We have already started to migrate resources in a ad-hoc fashion to v2, and generally the changes required are straightforward and predictable, however we have had to deal with some differences in behavior.

Currently all new resources are required to target v2, but existing resources are permitted to be enhanced while remaining on V1. We need to plan for the deprecation given bulk of resources in the provider are based on the older SDK.

#### Start of Maintenance Mode: July 31, 2024

Effectively this means that feature development on existing resources will require migration to v2 of the provider, as new features will not be added to that version off the SDK.

#### End of Support: July 31, 2025

By this time the provider should not have any functionality dependent on V1 of the SDK.

### Affected Resource(s) and/or Data Source(s)

All targeting V1

### References

- https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-aws-sdk-for-go-v1-on-july-31-2025/

**Comment by breathingdust** [comment]
Closing in favor of tracking work in #32976
